allennlp/data/dataset_readers/vqav2.py

```diff
@@ -0,0 +1,563 @@
+                self._coordinates_cache_instance = TensorCache(
+                    os.path.join(self.feature_cache_dir, "coordinates")
+                )
+
+        return self._coordinates_cache_instance
+
+    @overrides
+    def _read(self, split_name: str):
+        class Split(NamedTuple):
+            annotations: Optional[str]
+            questions: str
+
+        aws_base = "https://s3.amazonaws.com/cvmlp/vqa/"
+        mscoco_base = aws_base + "mscoco/vqa/"
+        scene_base = aws_base + "abstract_v002/vqa/"
+        splits = {
+            "balanced_real_train": Split(
+                mscoco_base
+                + "v2_Annotations_Train_mscoco.zip!v2_mscoco_train2014_annotations.json",
+                mscoco_base
+                + "v2_Questions_Train_mscoco.zip!v2_OpenEnded_mscoco_train2014_questions.json",
+            ),
+            "balanced_real_val": Split(
+                mscoco_base + "v2_Annotations_Val_mscoco.zip!v2_mscoco_val2014_annotations.json",
+                mscoco_base
+                + "v2_Questions_Val_mscoco.zip!v2_OpenEnded_mscoco_val2014_questions.json",
+            ),
+            "balanced_real_test": Split(
+                None,
+                mscoco_base
+                + "v2_Questions_Test_mscoco.zip!v2_OpenEnded_mscoco_test2015_questions.json",
+            ),
+            "balanced_bas_train": Split(  # "bas" is Binary Abstract Scenes
+                scene_base
+                + "Annotations_Binary_Train2017_abstract_v002.zip!abstract_v002_train2017_annotations.json",
+                scene_base
+                + "Questions_Binary_Train2017_abstract_v002.zip!OpenEnded_abstract_v002_train2017_questions.json",
+            ),
+            "balanced_bas_val": Split(
+                scene_base
+                + "Annotations_Binary_Val2017_abstract_v002.zip!abstract_v002_val2017_annotations.json",
+                scene_base
+                + "Questions_Binary_Val2017_abstract_v002.zip!OpenEnded_abstract_v002_val2017_questions.json",
+            ),
+            "abstract_scenes_train": Split(
+                scene_base
+                + "Annotations_Train_abstract_v002.zip!abstract_v002_train2015_annotations.json",
+                scene_base
+                + "Questions_Train_abstract_v002.zip!OpenEnded_abstract_v002_train2015_questions.json",
+            ),
+            "abstract_scenes_val": Split(
+                scene_base
+                + "Annotations_Val_abstract_v002.zip!abstract_v002_val2015_annotations.json",
+                scene_base
+                + "Questions_Val_abstract_v002.zip!OpenEnded_abstract_v002_val2015_questions.json",
+            ),
+            "abstract_scenes_test": Split(
+                None,
+                scene_base
+                + "Questions_Test_abstract_v002.zip!OpenEnded_abstract_v002_test2015_questions.json",
+            ),
+        }
+
+        if isinstance(split_name, list):
```

Comment:
@matt-gardner, in Python, strings are lists 🙄. I'll change this around.

Reply:
Oh man, that's unexpected...